clippy_lints/src/missing_const_for_fn.rs

```diff
@@ -149,6 +151,11 @@ impl<'tcx> LateLintPass<'tcx> for MissingConstForFn {
             return;
         }
 
+        // Skip if there is any pointer deref ops in the body, and the msrv of that feature does not meet.
+        if !self.msrv.meets(msrvs::CONST_RAW_PTR_DEREF) && has_raw_ptr_deref_in_body(cx, body) {
```

Comment:
Should this logic be in `is_min_const_fn` instead? Other lints also use that to check if something can be `const`:
```rs
thread_local! {
    static V: u8 = unsafe { *(&1u8 as *const u8) };
}
```
^ This probably also can't be made `const` for old MSRVs?

Reply:
well, I'm not familiar with those mir stuffs, so I don't know where should I put the equivalent logic in that function.

and that thread_local lint was added in `1.77` so I think it should be fine naturally~

Reply:
probably in this match https://github.com/rust-lang/rust-clippy/blob/master/clippy_utils/src/qualify_min_const_fn.rs#L270

`ProjectionElem::Deref` if `base.ty().is_unsafe_ptr()` should be it I think.